File: clientexercises/src/app/expense/expense.ts
/**
* expense - interface for expenses
*/
export interface Expense {
    id: number;
    employeeid: number;
    categoryid: string;
    description: string;
    receipt: boolean;
    dateincurred: string;
    amount: number;
    receiptscan: string | null | ArrayBuffer;
   }
   

File: exercises/src/main/java/com/info5059/exercises/expense/Expense.java
package com.info5059.exercises.expense;
import lombok.Data;
import lombok.RequiredArgsConstructor;
import java.math.BigDecimal;
import javax.persistence.Basic;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.Lob;
/**
* Expense entity
*/
@Entity
@Data
@RequiredArgsConstructor
public class Expense {
 @Id
 @GeneratedValue(strategy = GenerationType.IDENTITY)
 private long Id;
 private long employeeid; // FK
 private String categoryid; // FK
 private String description;
 private boolean receipt;
 private BigDecimal amount;
 private String dateincurred;
 // needed in 2nd case study
 @Basic(optional = true)
 @Lob
 private String receiptscan;
}
